### ISSUE TITLE
检查点恢复训练的时候，如果train_sampler is None，样本的顺序是不是就是没有shuffle的，和正常训练不一致！

### ISSUE THREAD

**Issue by github-luffy** [jingyaogong/minimind#645]
根据代码逻辑，这里可能存在不一致性：

正常训练：有shuffle控制
shuffle=(train_sampler is None) 
恢复训练：如果train_sampler为None，使用range(len(train_ds))

这意味着：
正常训练时（没有sampler）：会shuffle
恢复训练时（没有sampler）：不会shuffle

如果要保持一致性，应该修改SkipBatchSampler的创建：
if train_sampler is None:
    # 创建一个随机采样器
    random_sampler = torch.utils.data.RandomSampler(train_ds)
    batch_sampler = SkipBatchSampler(random_sampler, args.batch_size, start_step + 1)
else:
    batch_sampler = SkipBatchSampler(train_sampler, args.batch_size, start_step + 1)

大佬这样的设计，可能的原因：

恢复训练是特殊情况：可能认为恢复训练时顺序不重要

简化逻辑：使用range(len(train_ds))最简单

分布式训练为主：代码可能主要针对分布式场景（总有train_sampler）。我的理解有问题吗？

**Comment by jingyaogong** [comment]
https://github.com/jingyaogong/minimind/issues/644
